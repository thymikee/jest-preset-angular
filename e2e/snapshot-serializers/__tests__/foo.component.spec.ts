import { Component, Input } from '@angular/core';
import { TestBed, waitForAsync } from '@angular/core/testing';

@Component({
  selector: 'foo',
  templateUrl: './foo.component.html',
  styleUrls: ['./foo.component.scss'],
  // we have to setup styles this way, since simple styles/styleUrs properties will be removed (jest does not unit test styles)
  styles: [
    `
      p {
        color: red;
      }
    `,
  ],
})
class FooComponent {
  @Input() value1 = 'val1';
  @Input() value2 = 'val2';

  condition1 = true;
  condition2 = false;
}

test('snapshot should work', waitForAsync(() => {
  TestBed.configureTestingModule({
    declarations: [FooComponent],
  });

  const fixture = TestBed.createComponent(FooComponent);
  fixture.detectChanges();

  expect(fixture).toMatchSnapshot();
  expect(fixture.debugElement.nativeElement).toMatchSnapshot();
}));
