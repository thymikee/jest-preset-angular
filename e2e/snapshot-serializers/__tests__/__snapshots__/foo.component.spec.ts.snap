// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should work 1`] = `
<foo
  __ngContext__="0"
  condition1={[Function Boolean]}
  condition2="false"
  value1={[Function String]}
  value2={[Function String]}
>
  <p>
    Line 1
  </p><div>
    <div>
       val1 
    </div>
  </div>
</foo>
`;

exports[`snapshot should work 2`] = `
<div
  id="root0"
>
  <p>
    Line 1
  </p>
  <div>
    <div>
       val1 
    </div>
    
    
  </div>
</div>
`;
